tests/test_header_matching.js

```diff
@@ -91,34 +93,32 @@ test('match headers on number with regexp', async t => {
     headers: { 'X-My-Headers': 123 },
   })
 
-  t.equal(statusCode, 200)
-  t.equal(body, 'Hello World!')
+  expect(statusCode).to.equal(200)
+  expect(body).to.equal('Hello World!')
   scope.done()
 })
 
-test('match header on scope with function: gets the expected argument', async t => {
-  t.plan(3)
+test('match header on scope with function: gets the expected argument', async () => {
+  const matchHeaderSpy = sinon.stub().returns(false)
+  // TODO: It's surprising that this function receives a number instead of
+  // a string. Probably this behavior should be changed.
+  matchHeaderSpy.withArgs(456).returns(true)
```

Comment:
Hm, it seems like this test should assert this call is the one that’s made.

Reply:
yep, I missed that. updated.